CMakeLists.txt

```diff
@@ -361,13 +338,11 @@ ELSE(CMAKE_TOOLCHAIN_FILE)
   #------------------------------------
   # Identify the compiler
   #------------------------------------
-  IF( $ENV{CXX} MATCHES "icpc" )
-    SET( COMPILER Intel )
-  ELSEIF($ENV{CXX} MATCHES "xlC" OR CMAKE_C_COMPILER_ID MATCHES "XL")
-    SET( COMPILER IBM )
-  ELSEIF ( CMAKE_C_COMPILER_WORKS OR CMAKE_C_COMPILER_WORKS )
+  IF ( CMAKE_C_COMPILER_WORKS OR CMAKE_C_COMPILER_WORKS )
     IF( CMAKE_COMPILE_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX )
```

Comment:
While we are updating: I suggest updating this section to 

1.  Use CMAKE_C_COMPILER_ID and CMAKE_CXX_COMPILER_ID consistently, as per the Intel check, for GNU and IBM. See https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_ID.html
2. Remove the microsoft check. We know of no one using this in years and it is a totally unchecked configure path. We have an "unknown compiler" route.